src/go/k8s/internal/controller/redpanda/managed_decommission_controller.go

```diff
@@ -0,0 +1,580 @@
+		}
+
+		requestedReplicas = int(ptr.Deref(sts.Spec.Replicas, 0))
+	}
+
+	if requestedReplicas == 0 {
+		return 0, ErrZeroReplicas
+	}
+	return requestedReplicas, nil
+}
+
+func podEvict(ctx context.Context, l logr.Logger, c k8sclient.Client, rp *v1alpha1.Redpanda) error {
+	if rp.Status.ManagedDecommissioningNode == nil {
+		return nil
+	}
+
+	decommissionNodeID := int(*rp.Status.ManagedDecommissioningNode)
+	log := l.WithName("podEvict").WithValues("decommission-node-id", decommissionNodeID)
+
+	pod, err := getPodFromRedpandaNodeID(ctx, l, c, rp, decommissionNodeID)
+	if err != nil {
+		return fmt.Errorf("getting pod from node-id (%d): %w", decommissionNodeID, err)
+	}
+
+	log.Info("delete Pod PVC", "pod-name", pod.Name)
+	if err = utils.DeletePodPVCs(ctx, c, pod, log); err != nil {
+		return fmt.Errorf(`unable to remove VPCs for pod "%s/%s: %w"`, pod.GetNamespace(), pod.GetName(), err)
+	}
+	return c.Delete(ctx, pod)
+}
+
+func getPodFromRedpandaNodeID(ctx context.Context, l logr.Logger, c k8sclient.Client, rp *v1alpha1.Redpanda, nodeID int) (*corev1.Pod, error) {
+	log := l.WithName("getPodFromRedpandaNodeID")
+
+	pl, err := getPodList(ctx, c, rp.Namespace, labels.Set{
+		"app.kubernetes.io/component": componentLabelValue,
+		"app.kubernetes.io/instance":  rp.Name,
+	}.AsSelector())
+	if err != nil {
+		return nil, fmt.Errorf("check all pods have condition: %w", err)
+	}
+
+	valuesMap, err := getHelmValues(l, rp.GetHelmReleaseName(), rp.Namespace)
+	if err != nil {
+		return nil, fmt.Errorf("get helm values: %w", err)
+	}
+
+	for i := range pl.Items {
+		pod := pl.Items[i]
+
+		log.WithValues("pod-name", pod.Name)
+
+		if !utils.IsPodReady(&pod) {
+			log.Info("pod is not ready",
+				"containersReady", utils.IsStatusPodConditionTrue(pod.Status.Conditions, corev1.ContainersReady),
+				"podReady", utils.IsStatusPodConditionTrue(pod.Status.Conditions, corev1.PodReady),
+				"pod-phase", pod.Status.Phase)
+			return nil, &resources.RequeueAfterError{
+				RequeueAfter: wait.Jitter(decommissionWaitInterval, decommissionWaitJitterFactor),
+				Msg:          "pod is not ready",
+			}
+		}
+
+		var ordinal int32
+		ordinal, err := utils.GetPodOrdinal(pod.Name, pod.GenerateName[:len(pod.GenerateName)-1])
+		if err != nil {
+			return nil, err
+		}
+		podOrdinal := int(ordinal)
+
+		log.WithValues("pod-ordinal", ordinal)
+
+		singleNodeAdminAPI, err := buildAdminAPI(rp.GetHelmReleaseName(), rp.Namespace, 0, &podOrdinal, valuesMap)
+		if err != nil {
+			return nil, fmt.Errorf("creating single node AdminAPI for pod (%d): %w", podOrdinal, err)
+		}
+
+		nodeCfg, err := singleNodeAdminAPI.GetNodeConfig(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("getting node configuration from pod (%d): %w", podOrdinal, err)
+		}
+
+		if nodeCfg.NodeID == nodeID {
+			return &pod, nil
+		}
+	}
+
+	return nil, nil
+}
+
+func getBrokerByBrokerID(ctx context.Context, brokerID int, adminAPI *admin.AdminAPI) (*admin.Broker, error) {
```

Comment:
Is brokerID the same thing as Pod Ordinal? the naming is a little confusing.

Reply:
Interesting this function is unused, so I will remove it.